zerver/views/message_fetch.py

```diff
@@ -106,19 +106,40 @@ def get_messages_backend(
     *,
     anchor_val: Annotated[str | None, ApiParamConfig("anchor")] = None,
     include_anchor: Json[bool] = True,
-    num_before: Json[NonNegativeInt],
-    num_after: Json[NonNegativeInt],
+    num_before: Json[NonNegativeInt] = 0,
+    num_after: Json[NonNegativeInt] = 0,
     narrow: Json[list[NarrowParameter] | None] = None,
     use_first_unread_anchor_val: Annotated[
         Json[bool], ApiParamConfig("use_first_unread_anchor")
     ] = False,
     client_gravatar: Json[bool] = True,
     apply_markdown: Json[bool] = True,
+    message_ids_to_fetch: Annotated[
+        Json[list[NonNegativeInt] | None], ApiParamConfig("message_ids")
+    ] = None,
 ) -> HttpResponse:
+    if message_ids_to_fetch is None:
+        message_ids_to_fetch = []
+    # User requested a list of message ids to fetch.
+    is_message_ids_pre_specified = len(message_ids_to_fetch) > 0
```

Comment:
Might be good to integrate this with the `message_ids_to_fetch` None check above; depending on the answer to https://chat.zulip.org/#narrow/stream/378-api-design/topic/Message.20matches.20narrow/near/1951550, we might want to handle no parameter vs. empty parameter differently. 

I think we can also do the following names:
- `message_ids_to_fetch` => `client_requested_message_ids` -- makes really clear these aren't just some internal value partway through the normal fetching process.
- `is_message_ids_pre_specified` => Do we need this, vs. just checking `if client_requested_message_ids is not None` and setting the value for that properly?

Reply:
Removed `is_message_ids_pre_specified` and did the rename.